test/integration/compile-swagger-test.coffee

```diff
@@ -97,25 +97,44 @@ describe('compile() · Swagger', ->
   )
 
   describe('with multiple responses', ->
-    # Multiple responses is the closest we can get with Swagger to something
-    # like transaction examples ¯\_(ツ)_/¯
-
+    transactions = undefined
+    filename = 'apiDescription.json'
     detectTransactionExamples = sinon.spy(require('../../src/detect-transaction-examples'))
-    compilationResult = undefined
-    transaction = undefined
+
+    expectedStatusCodes = [200, 400, 500]
 
     beforeEach((done) ->
       stubs = {'./detect-transaction-examples': detectTransactionExamples}
 
-      compileFixture(fixtures.multipleResponses.swagger, {stubs}, (args...) ->
+      compileFixture(fixtures.multipleResponses.swagger, {filename, stubs}, (args...) ->
         [err, compilationResult] = args
-        transaction = compilationResult.transactions[0]
+        transactions = compilationResult?.transactions
         done(err)
       )
     )
 
     it('detection of transaction examples was not called', ->
       assert.isFalse(detectTransactionExamples.called)
     )
+    it('expected number of transactions was returned', ->
+      assert.equal(transactions.length, expectedStatusCodes.length)
+    )
+
+    for statusCode, i in expectedStatusCodes
+      do (statusCode, i) ->
+        context("origin of transaction ##{i + 1}", ->
+          it('uses URI as resource name', ->
+            assert.equal(transactions[i].origin.resourceName, '/honey')
+          )
+          it('uses method as action name', ->
+            assert.equal(transactions[i].origin.actionName, 'GET')
```

Comment:
I don't see how just having a variable helps. The best would be having a
variable which is used to build the fixture and then expected in assert.
That's less easy to write, but IMHO makes the best sense and makes the test
better in every aspect. However, just moving a literal to a variable, which
still isn't tied to the actual fixture in any way, is probably just a
matter of preference.

On 8 Sep 2016 2:41 p.m., "Mila Votradovec" <notifications@github.com> wrote:

> In test/integration/compile-swagger-test.coffee
> <https://github.com/apiaryio/dredd-transactions/pull/63#discussion_r77997366>
> :
>
> >          done(err)
> >        )
> >      )
> >
> >      it('detection of transaction examples was not called', ->
> >        assert.isFalse(detectTransactionExamples.called)
> >      )
> > +    it('expected number of transactions was returned', ->
> > +      assert.equal(transactions.length, expectedStatusCodes.length)
> > +    )
> > +
> > +    for statusCode, i in expectedStatusCodes
> > +      do (statusCode, i) ->
> > +        context("origin of transaction ##{i + 1}", ->
> > +          it('uses URI as resource name', ->
> > +            assert.equal(transactions[i].origin.resourceName, '/honey')
> > +          )
> > +          it('uses method as action name', ->
> > +            assert.equal(transactions[i].origin.actionName, 'GET')
>
> Although I understand your point, I would dispute it with a fact that
> reviewer is unable to verify expected data unless he would:
>
>    1. Find where the fixtures is loaded
>    2. Find the source of the fixtures
>    3. Goes to the fixture file to verify the literal
>
> Declaring variable with expected value near the place, where the fixture
> is required would omit first two steps and saves more brain cycles, IMHO :-)
>
> With current setup, I have to convince myself to make a journey to verify
> the expected value, which could lead to skipping typos (goat vs boat;
> both can be legit) in false positive test.
>
> —
> You are receiving this because you modified the open/close state.
> Reply to this email directly, view it on GitHub
> <https://github.com/apiaryio/dredd-transactions/pull/63/files/a1c7007fd63b348eff31086d0866d2ad5538cd38#r77997366>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AARTMdrwo_-sQx8fSwL7qzP-jDo3uNcMks5qoAKOgaJpZM4Js3K4>
> .
>